scipy/integrate/_quadpack_py.py

```diff
@@ -281,6 +281,91 @@ def quad(func, a, b, args=(), full_output=0, epsabs=1.49e-8, epsrel=1.49e-8,
+
+    The following provides a short desciption from [1]_ for each
+    routine.
+
+    qagse
+        is an integrator based on globally adaptive interval
+        subdivision in connection with extrapolation, which will
+        eliminate the effects of integrand singularities of
+        several types.
+    qagie
+        handles integration over infinite intervals. The infinite range is
+        mapped onto a finite interval and subsequently the same strategy as
+        in QAGS is applied.
+    qagpe
+        serves the same purposes as QAGS, but also allows the
+        user to provide explicit information about the location
+        and type of trouble-spots i.e. the abscissae of internal
+        singularities, discontinuities and other difficulties of
+        the integrand function.
+    qawoe
+        is an integrator for the evaluation of
+        :math:`\\int^b_a \\cos(\\omega x)f(x)dx` or
+        :math:`\\int^b_a \\sin(\\omega x)f(x)dx`
+        over a finite interval [a,b], where :math:`\\omega` and :math:`f`
+        are specified by the user. The rule evaluation component is based
+        on the modified Clenshaw-Curtis technique
+
+        An adaptive subdivision scheme is used in connection
+        with an extrapolation procedure, which is a modification
+        of that in QAGS and allows the algorithm to deal with
+        singularities in :math:`f(x)`.
+    qawfe
+        calculates the Fourier transform
+        :math:`\\int^\\infty_a \\cos(\\omega x)f(x)dx` or
+        :math:`\\int^\\infty_a \\sin(\\omega x)f(x)dx`
+        for user-provided :math:`\\omega` and :math:`f`. The procedure of QAWO
+        is applied on successive finite intervals, and convergence
+        acceleration by means of the :math:`\\varepsilon`-algorithm is applied
+        to the series of integral approximations.
+    qawse
+        approximate :math:`\\int^b_a w(x)f(x)dx`, with :math:`a < b` where
+        :math:`w(x) = (x-a)^{\\alpha}(b-x)^{\\beta}v(x)` with
+        :math:`\\alpha,\\beta > -1`, where :math:`v(x)` may be one of the
+        following functions: :math:`1`, :math:`\\log(x-a)`, :math:`\\log(b-x)`,
+        :math:`\\log(x-a)\\log(b-x)`.
+
+        The user specifies :math:`\\alpha`, :math:`\\beta` and the type of the
+        function :math:`v`. A globally adaptive subdivision strategy is
+        applied, with modified Clenshaw-Curtis integration on those
+        subintervals which contain `a` or `b`.
+    qawce
+        compute :math:`\\int^b_a f(x) / (x-c)dx` where the integral must be
+        interpreted as a Cauchy principal value integral, for user specified
+        :math:`c` and :math:`f`. The strategy is globally adaptive. Modified
+        Clenshaw-Curtis integration is used on those intervals containing the
+        point :math:`x = c`.
+
+    References
+    ----------
+
+    .. [1] Piessens, Robert; de Doncker-Kapenga, Elise;
+           Überhuber, Christoph W.; Kahaner, David (1983).
```

Comment:
We only allow a set of utf8 chars
```suggestion
           Uberhuber, Christoph W.; Kahaner, David (1983).
```

Reply:
We're conservative about adding arbitrary Unicode characters, but Ü (and any other upper- or lower-case characters that appear in names) should be acceptable.

Reply:
@WarrenWeckesser I don't know the rational behind this. Is this still a problem on some platform or are is this a legacy thing?

Reply:
It started out with Python 2, where the default Python file encoding was ASCII.  Occasionally someone would copy-paste text (typically a reference) from, say, a PDF document into a docstring, and it would turn out that the copied text contained some non-ASCII characters that were not obvious from looking at the text.  (I seem to recall alternative hyphen characters showing up more than once.)  With Python 2, these characters would prevent Python from accepting the code, but the error message wouldn't always make clear exactly which character was causing the problem.  The script `unicode-check.py` started out as a script to print the exact character (and its location) that was not ASCII.

With Python 3, the default encoding is UTF-8, so there is no longer a *requirement* to avoid non-ASCII characters.  We've been adding Unicode characters to the "allow" list as needed.  I guess it is probably as good a time as any to now allow, say, any printable Latin1 (i.e. ISO-8859-1) letters, instead of continuing to add the various characters with accents, umlauts, circumflexes, etc., as they come up in PRs,   However, I think we should still avoid the so-called ["confusable" Unicode characters](https://util.unicode.org/UnicodeJsps/confusables.jsp).  Also, it turns out that the Unicode check helps avoid the ["trojan source" security threat](https://trojansource.codes/trojan-source.pdf).

I'll create a PR to update `unicode-check.py`, and we can figure out the extent to which we want to expand the allowable character set there.

Reply:
PR is https://github.com/scipy/scipy/pull/16060